pytorch_tabnet/tab_model.py

```diff
@@ -37,111 +61,20 @@ def __init__(self,
-        values set in **kwargs.
-
-        Parameters
-        ----------
-            loss_fn: :class: `torch.nn.functional`
-                The loss function. A few options:
-                    - torch.nn.functional.mse_loss
-                    - torch.nn.functional.binary_cross_entropy
-                    - torch.nn.functional.cross_entropy
-                    - torch.nn.functional.l1_loss
-            max_epochs: int
-                The maximum number of epochs for training
-            patience: int
-            learning_rate: float
-                The initial learning rate
-            schedule: str
-                The learning rate schedule('lambda', 'cos',
-                'exp' or 'step' or None).
-            lr_params: dict
-                Additional infos on the learning rate scheduler
-                See https://pytorch.org/docs/master/optim.html for params
-                of each scheduler type.
-            optimizer_fn: :class: torch.optim object
-                The optimizer function to be used. A few options:
-                    - torch.optim.SGD
-                    - torch.optim.Adam
-                    - torch.optim.Adadelta
-            opt_params: dict
-                Further parameters to be used by optimizer_fn.
-                or None)
-
-        """
-        # default params
-        self.max_epochs = 100
-        self.patience = 15
-        self.lr = 2e-2
-        self.scheduler = None
-        self.lr_params = {}
-        self.opt_params = {}
-        self.optimizer_fn = None
-        self.clip_value = None
-        self.model_name = "DQTabNet"
-        self.lambda_sparse = 1e-3
-        self.scheduler_fn = None
-        self.patience_counter = 0
-        self.batch_size = 1024
-        self.saving_path = "./"
-        self.verbose = 1
-
-        # Overrides parameters
-        self.__dict__.update(kwargs)
-
-        self.output_dim = self.network.output_dim
-
-        if self.output_dim == 1:
-            self.loss_fn = torch.nn.functional.mse_loss
-        else:
-            self.loss_fn = torch.nn.functional.cross_entropy
-
-        self.opt_params['lr'] = self.lr
-
-        if self.optimizer_fn is None:
-            self.optimizer = torch.optim.Adam(self.network.parameters(),
-                                              **self.opt_params)
-        else:
-            self.optimizer = self.optimizer_fn(self.network.parameters(),
-                                               **self.opt_params)
-
-        if self.scheduler_fn:
-            self.scheduler = self.scheduler_fn(self.optimizer, **self.scheduler_params)
+        Returns
+        -------
+        train_dataloader, valid_dataloader : torch.DataLoader, torch.DataLoader
+            Training and validation dataloaders
+        -------
+        """
+        raise NotImplementedError('users must define construct_loaders to use this base class')
 
     def fit(self, X_train, y_train, X_valid=None, y_valid=None,
```

Comment:
Shouldn't we call `load_best_model` at the end of this function? I'm not sure this is the best solution, but it can be misleading if the output model after the fit is completed is not the one that performs best on the validation set.

Reply:
I agree added it!